h/static/scripts/components/AnnotationCard.tsx

```diff
@@ -30,14 +30,38 @@ import { callAPI } from '../util/api';
 import { APIError } from '../util/api';
 import AnnotationDocument from './AnnotationDocument';
 
+export type StatusChangeResult =
+  | {
+      status: 'saving';
+    }
+  | {
+      status: 'saved';
+      newModerationStatus: ModerationStatus;
+    }
+  | {
+      status: 'error';
+      error: unknown;
+    };
```

Comment:
I'm not convinced with the `status` name for the discriminator prop, as it can be confusing in combination with `newModerationStatus`, but I couldn't come up with a better naming.

I'm open to suggestions.